### PR TITLE
[Develop] new component Table

### PR DESCRIPTION
**Pull request by naimdanialjohar**
# Description

new component Table.
also included in this change;
- `Button` added class for link buttons
- `Checkbox` removed `flex` because it caused alignment issues
- `List` removed unused `props` from React.FC
- `Modal` removed unused `className` prop
- `Pagination` export `itemRender`, used in `Table` component for `pagination` prop
- `Tree` created only to edit the `style.css` which is used in `Table` component

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Checkbox/style.css [hunk 1]
```diff
@@ -8,7 +8,7 @@
   }
 
   &-wrapper {
-    @apply text-slate-700 font-inter font-normal flex;
+    @apply text-slate-700 font-inter font-normal;
```
Comment (AladdinNullptr) [inline src/Checkbox/style.css:11]: Did you make sure this didn't break anything?

File: src/Table/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { ColumnType } from 'antd/lib/table';
+import { SortOrder } from 'antd/lib/table/interface';
+import { ArrowsDownUp, CaretLeft, CaretRight, Funnel } from 'phosphor-react';
+import { itemRender } from '../Pagination';
+
+import './style.css';
+
+export type TableProps<T> = AntTableProps<T>;
+export const FilterIcon = <Funnel size={16} weight="light" />;
+export const headerWithSort = <RecordType extends {}>(
```
Comment (AladdinNullptr) [inline src/Table/index.tsx:12]: Should be `HeaderWithSort`

File: src/Table/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  sortColumns: {
+    column: ColumnType<RecordType>;
+    order: SortOrder;
+  }[],
+  title: string
```
Comment (AladdinNullptr) [inline src/Table/index.tsx:17]: type should be React Node

File: src/Checkbox/style.css (hunk 1 above)
Comment (naimdanialjohar) [reply src/Checkbox/style.css:11]: @AladdinNullptr yes i checked the `Checkbox` component and other components that use `Checkbox`. nothing breaks.

File: src/Table/style.css [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+@import '../Input/style.css';
+@import '../Tree/style.css';
+
+.ant-table-wrapper {
+  .ant-table {
+    @apply font-inter;
+    @apply rounded-xl;
+    @apply overflow-hidden;
+    @apply border;
+    @apply border-slate-100;
```
Comment (AladdinNullptr) [inline src/Table/style.css:20]: can put in one or two `apply`

File: src/Table/style.css [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    .ant-table-thead {
+      tr {
+        th {
+          @apply bg-transparent;
+          @apply font-semibold;
+          @apply text-sm;
+          @apply text-neutral-700;
+          @apply rounded-tr-xl;
```
Comment (AladdinNullptr) [inline src/Table/style.css:43]: can put in one or two `apply`

File: src/Table/style.css [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+        &.ant-table-cell-with-append {
+          @apply pl-10;
+          svg {
+            @apply absolute;
+            @apply m-auto;
+            @apply top-0;
+            @apply bottom-0;
+            @apply left-3;
+            @apply transition-all;
+            @apply cursor-pointer;
```
Comment (AladdinNullptr) [inline src/Table/style.css:126]: can put in one or two `apply`

File: src/Table/style.css [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+.ant-table-filter-dropdown {
+  @apply shadow-md;
+  @apply rounded-lg;
+
+  .ant-table-filter-dropdown-tree {
+    @apply pt-3;
+    @apply pr-3;
+    @apply pb-0;
+    @apply pl-3;
```
Comment (AladdinNullptr) [inline src/Table/style.css:176]: can put in one `apply`

Comment (AladdinNullptr) [reply src/Table/style.css:176]: also maybe can do `p-3 pb-0`

File: src/Table/style.css [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+  }
+
+  .ant-table-filter-dropdown-btns {
+    @apply border-none;
+    @apply justify-end;
+    @apply rounded-b-lg;
+    @apply p-2;
```
Comment (AladdinNullptr) [inline src/Table/style.css:183]: can put in one or two `apply`

File: src/Table/style.css [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+.row-dragging {
+  @apply font-inter;
+  @apply flex;
+  @apply items-center;
+  @apply bg-neutral-100;
+  @apply border;
+  @apply border-slate-100;
+  /* hardcoded to none in antd codebase */
+  @apply pointer-events-auto !important;
+  @apply cursor-grabbing;
```
Comment (AladdinNullptr) [inline src/Table/style.css:197]: can put in one or two `apply`

File: src/Table/style.css [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+    @apply relative;
+
+    svg {
+      @apply absolute;
+      @apply m-auto;
+      @apply top-0;
+      @apply bottom-0;
+      @apply left-3;
+      @apply transition-all;
+      @apply cursor-pointer;
```
Comment (AladdinNullptr) [inline src/Table/style.css:229]: can put in one or two `apply`

**Review by AladdinNullptr — APPROVED** [review]

**Pull request merged**